src/main/java/uk/gov/hmcts/reform/mi/micore/model/CoreCaseData.java

```diff
@@ -1,13 +1,15 @@
 package uk.gov.hmcts.reform.mi.micore.model;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.AllArgsConstructor;
 import lombok.Getter;
 
 import java.util.Map;
 
 /**
  * Base model for CoreCaseData read from Data Extractor output.
  */
+@AllArgsConstructor
```

Comment:
you can use @Value and @Builder for lombok immutable model definition

Reply:
Made change